rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
From the rmw thread it sounds like we agree that a null rmw_time struct should indicate blocking forever.

Should the user pass a null pointer to a `std::duration` to block forever, or should a negative `std::duration` block forever?